addon/content/modules/contact.js

```diff
@@ -182,20 +127,44 @@ ContactFromAB.prototype = {
   },
 
   /**
-   * The aEmail parameter is here because the same contact object is shared for
+   * The `email` parameter is here because the same contact object is shared for
    * all instances of a contact, even though the original email address is
    * different. This allows one to share a common color for a same card in the
    * address book.
    */
-  toTmplData(useColor, position, email, isDetail) {
-    let [name, extra] = this.getName(position, isDetail);
-    let displayEmail = name != email ? email : "";
-    let hasCard = this._card != null;
-    let skipEmail =
+  async toTmplData(useColor, position, email, isDetail) {
+    const allIdentities = await browser.convContacts
+      .getIdentities({ includeNntpIdentities: true })
+      .catch(console.error);
+    const identities = await browser.convContacts
+      .getIdentities({ includeNntpIdentities: false })
+      .catch(console.error);
+    const identity = allIdentities.find(
+      ident => ident.identity.email.toLowerCase() == this._email.toLowerCase()
+    );
+    const multipleIdentities = identities.length > 1;
```

Comment:
I think the `includeNntpIdentities` parameteres are the wrong way round, but looking at the old code, I think we can simplify it anyway.

In the old code, `getIdentityForEmail` passes false as the parameter and so having `getIdentities().length > 1` default to true is a little weird - as it'd never be hit unless there was at least one identity for the email.

So I think we can only get the identities once, using `includeNntpIdentities: false`, and drop the multipleIdentities check, e.g. `extra = this._email`

Reply:
I don't quite understand this (nor have I used NNTP in the last 10 years....), but I'll drop the checks as per your suggestion.

Reply:
NNTP is a separate account type that Conversations doesn't support (because gloda, the global database, doesn't), so it is unlikely that we're going to be matching one of the news identities.